internal/pkg/encoding/jsonnet/vm.go

```diff
@@ -16,13 +16,13 @@ type VM[T any] struct {
 	lock    sync.Mutex
 	vm      *jsonnet.VM
 	err     error
-	payload T
+	payload *T
 }
 
 // GetPayload returns the payload of the current evaluation.
 // Can be used by custom functions for request specific values.
 func (vm *VM[T]) GetPayload() T {
-	return vm.payload
+	return *vm.payload
```

Comment:
You should return `*T` if you want to keep original logic, and don't copy the payload.

------------

If you keep `VM[*receive.Context]`, the payload type is `**receive.Context` and it is a little bit weird.
So `VM[receive.Context]` seem to me as a better solution.
```
T = receive.Context
*T = *receive.Context
VM[*receive.Context] - > VM[receive.Context]`
Evaluate(T) -> Evaluate(*T)
Payload() T -> Payload() *T
```

Reply:
:heavy_check_mark: :+1: